core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -94,10 +101,24 @@ public void aggregate(String groupKey, Iterator<StructuredRecord> iterator,
     builder.set(conf.getKeyField(), groupKey);
     while (iterator.hasNext()) {
       StructuredRecord record = iterator.next();
-      String fieldName = record.get(conf.getFieldName());
-      fieldName = outputMappings.containsKey(fieldName) ? outputMappings.get(fieldName) : fieldName;
-      if (outputFields.contains(fieldName)) {
-        builder.set(fieldName, record.get(conf.getFieldValue()));
+      String outputFieldName = record.get(fieldName);
+      String outputFieldValue = record.get(fieldValue);
+
+      if (outputFieldName == null) {
+        if (!record.getSchema().getField(fieldName).getSchema().isNullable()) {
```

Comment:
also, whenever you see

```
if (x) {
  if (y) {
    do something
  }
}
```

it can be simplified to:
```
if (x && y) {
  do something
}
```

most IDEs will mark this as a warning.